edge-modules/DirectMethodSender/Settings.cs

```diff
@@ -25,22 +26,38 @@ public class Settings
                     configuration.GetValue<string>("TargetModuleId", "DirectMethodReceiver"),
                     configuration.GetValue<TransportType>("TransportType", TransportType.Amqp_Tcp_Only),
                     configuration.GetValue<TimeSpan>("DirectMethodDelay", TimeSpan.FromSeconds(5)),
-                    Option.Maybe(configuration.GetValue<Uri>("AnalyzerUrl")));
+                    Option.Maybe(configuration.GetValue<Uri>("AnalyzerUrl")),
+                    configuration.GetValue("InvocationSource", "Local"),
+                    Option.Maybe<string>(configuration.GetValue<string>("ServiceClientConnectionString")));
             });
 
         Settings(
             string deviceId,
             string targetModuleId,
             TransportType transportType,
             TimeSpan directMethodDelay,
-            Option<Uri> analyzerUrl)
+            Option<Uri> analyzerUrl,
+            string invocationSource,
+            Option<string> serviceClientConnectionString)
         {
             this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
             this.TargetModuleId = Preconditions.CheckNonWhiteSpace(targetModuleId, nameof(targetModuleId));
             Preconditions.CheckArgument(TransportType.IsDefined(typeof(TransportType), transportType));
             this.TransportType = transportType;
             this.DirectMethodDelay = Preconditions.CheckNotNull(directMethodDelay);
-            this.AnalyzerUrl = analyzerUrl;
+            this.AnalyzerUrl = Preconditions.CheckNotNull(analyzerUrl);
```

Comment:
Option type is struct; don't need to check for not null.

Reply:
Woops, I mis-merge this when I tried to resolve conflict.